cli/tests/environment-pull.bats

```diff
@@ -181,3 +215,47 @@ function update_dummy_env() {
   run "$FLOX_BIN" pull --remote floxtest/default
   assert_success
 }
+
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=pull:unsupported
+# pulling an environment without packages for the current platform
+#should fail with an error
+@test "l?: pull environment without packages for the current platform fails" {
+  make_dummy_env "owner" "name"
+  update_dummy_env "owner" "name"
+  make_incompatible "owner" "name"
+
+  run "$FLOX_BIN" pull --remote owner/name
+  assert_failure
+  assert_output --partial "This environment is not yet compatible with your system ($NIX_SYSTEM)"
+
+  run "$FLOX_BIN" pull --remote owner/name --add-system
+  assert_success
+}
+
+# bats test_tags=pull:unsupported:prompt-fail
+# pulling an environment without packages for the current platform
+#should fail with an error
+@test "l?: pull environment without packages for the current platform prompts for about adding system" {
+  update_dummy_env "owner" "name"
```

Comment:
question nonblocking: is this relying on the previous test calling `make_dummy_env`?

Reply:
`make_dummy_env` is called during setup()